### PR TITLE
🐛 Source Github: add custom pagination size for large streams

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
# Bug-fix Request

Fixes https://github.com/airbytehq/oncall/issues/75*

---

## Reason
*Reduced pagination size for heavily loaded streams, as this can be a problem for reading these streams using the API (the case was in the PR #4708 where the page size values were already reduced for large streams).
I also added a parameter to the specification that allows the user to configure this parameter depending on the degree of load of his repositories*

## Confirmation
 - [ ] **Was reproduced locally**

*Described above*

## How does the code change in the PR fix the issue?
*Described above*

---

### Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the new connector version is published, connector version bumped in the seed directory as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

### CODE REVIEW

File: airbyte-integrations/connectors/source-github/source_github/streams.py [hunk 1]
```diff
@@ -13,22 +13,27 @@
 from airbyte_cdk.sources.streams.http import HttpStream, HttpSubStream
 from requests.exceptions import HTTPError
 
+DEFAULT_PAGE_SIZE = 100
```
Comment (sergei-solonitcyn) [inline airbyte-integrations/connectors/source-github/source_github/streams.py:16]: Why do you have different default value here and in `spec.json`?

File: airbyte-integrations/connectors/source-github/source_github/streams.py [hunk 2]
```diff
@@ -295,6 +300,9 @@ class Organizations(GithubStream):
     API docs: https://docs.github.com/en/rest/reference/orgs#get-an-organization
     """
 
+    # GitHub pagination could be from 1 to 100.
+    page_size = 100
```
Comment (sergei-solonitcyn) [inline airbyte-integrations/connectors/source-github/source_github/streams.py:304]: And another one "page size" here. Is it something different?

File: airbyte-integrations/connectors/source-github/source_github/streams.py (hunk 1 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-github/source_github/streams.py:16]: This is the page size for all streams. Previously, we specified it in the parent stream (100) and redefined it for the child streams (4 large streams with a page size of 30/50), I simplified it a bit, now everything is specified in the parent type.

File: airbyte-integrations/connectors/source-github/source_github/streams.py (hunk 2 above)
Comment (yevhenii-ldv) [reply airbyte-integrations/connectors/source-github/source_github/streams.py:304]: Organizations streams work as a slightly separate branch, the `__init__` method is overridden there, and there are no large streams for Organizations, so we can safely use 100 as the page size.

**Review by sergei-solonitcyn — APPROVED** [review]

**Pull request merged**